Source/WebKit2/Platform/IPC/Connection.cpp

```diff
@@ -873,6 +873,7 @@ void Connection::didFailToSendSyncMessage()
     if (!m_shouldExitOnSyncMessageSendFailure)
         return;
 
+    m_client.didClose(*this);
```

Comment:
I didn't say anything about threads, but about calling didClose when the connection has already been invalidated. There's no problem with threads here, because didFailToSendSyncMessage() si only called from sendSyncMessage() that should always be called from the main thread. This is indeed a tricky issue, that's why it has been present since we switched to WebKit2 and we haven't fixed it yet :-)